src/ucp/tag/eager_rcv.c

```diff
@@ -86,10 +91,14 @@ ucp_eager_handler(void *arg, void *data, size_t length, void *desc,
 
```

Comment:
Maintainability: it is not obvious that data has been copied  in ucp_recv_desc_get. It will be easily broken when ucp_recv_desc_get is changed for instance. You could make a copy in the desc_get itself instead